src/main/java/taxiapp/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,43 @@
+package taxiapp.dao;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import taxiapp.db.Storage;
+import taxiapp.lib.Dao;
+import taxiapp.model.Manufacturer;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        Storage.addManufacturer(manufacturer);
+        return Storage.manufacturers.get(Storage.manufacturers.size() - 1);
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long manufacturerId) {
+        return Storage.manufacturers.stream()
+                .filter(m -> Objects.equals(manufacturerId, m.getId()))
+                .findFirst();
+    }
+
+    @Override
+    public List<Manufacturer> getAllManufacturer() {
+        return Storage.manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        Manufacturer oldValue = get(manufacturer.getId()).orElseThrow();
```

Comment:
```suggestion
        Manufacturer oldValue = get(manufacturer.getId()).get();
```
you can simply use get() if you don't want to throw a specific exception here